### PR TITLE
Added psnr/ssim evaluation function (replaces #15)

### PR DESCRIPTION
**Pull request by tmke8**
Fixed elements of the training script.

### CODE REVIEW

File: optimisation/training.py [hunk 1]
```diff
@@ -24,8 +26,8 @@ def train(args, train_loader, model, criterion, optimizer, epoch, summary_writer
             # Send inputs to correct device
             noisy = noisy.cuda() if args.cuda else noisy
             clean = clean.cuda() if args.cuda else clean
-            # ISO needs to be a 3d tensor to be passed to Gated Convolutions
-            iso = iso.view(noisy.size(0), -1, 1)
+
+            iso = torch.Tensor(iso)
```
Comment (Stoffle) [inline optimisation/training.py:30]: `iso` should already be a tensor I believe, was the `.view()` not necesssary?

File: optimisation/training.py [hunk 2]
```diff
@@ -40,21 +42,22 @@ def train(args, train_loader, model, criterion, optimizer, epoch, summary_writer
...
             # Update meters
-            loss.add(loss.item())
-            batch_time.add(time.time() - end)
+            loss_meter.add(loss.item())
+            batch_time_meter.add(time.time() - end)
             end = time.time()
 
             # Update progress bar
-            pbar.set_postfix(loss=loss.item())
+            pbar.set_postfix(loss=loss_meter.mean)
```
Comment (tmke8) [inline optimisation/training.py:50]: I think this might be wrong

File: optimisation/training.py [hunk 3]
```diff
@@ -90,30 +93,85 @@ def validate(args, val_loader, model, criterion, training_iters, summary_writer)
...
                 # Update meters
-                loss.add(loss.item())
-                batch_time.add(time.time() - end)
+                loss_meter.add(loss.item())
+                batch_time_meter.add(time.time() - end)
                 end = time.time()
 
                 # Update progress bar
-                pbar.set_postfix(loss=loss.item())
+                pbar.set_postfix(loss=loss_meter.mean)
```
Comment (tmke8) [inline optimisation/training.py:110]: This as well

File: optimisation/training.py (hunk 2 above)
Comment (Stoffle) [reply optimisation/training.py:50]: Depends if we want a running mean on the progress bar, or the loss for that batch - I think the mean would be more useful

File: optimisation/training.py (hunk 1 above)
Comment (tmke8) [reply optimisation/training.py:30]: Myles changed the code so that it doesn't need to be 3-dimensional anymore, I think
